nifty-core/src/main/java/com/facebook/nifty/ssl/NettyTcNativeLoader.java

```diff
@@ -0,0 +1,211 @@
+            throw (Error) new UnsatisfiedLinkError(
+                    "failed to load the required native library").initCause(UNAVAILABILITY_CAUSE);
+        }
+    }
+
+    private static void loadTcNative() {
+        String os = normalizeOs(System.getProperty("os.name", ""));
+        String arch = normalizeArch(System.getProperty("os.arch", ""));
+
+        Set<String> libNames = new LinkedHashSet<String>(3);
+        // First, try loading the platform-specific library. Platform-specific
+        // libraries will be available if using a tcnative uber jar.
+        libNames.add("netty-tcnative-" + os + '-' + arch);
+        if (LINUX.equalsIgnoreCase(os)) {
+            // Fedora SSL lib so naming (libssl.so.10 vs libssl.so.1.0.0)..
+            libNames.add("netty-tcnative-" + os + '-' + arch + "-fedora");
+        }
+        // finally the default library.
+        libNames.add("netty-tcnative");
+
+        boolean loaded = false;
+        Throwable lastException = null;
+        for (String libName : libNames) {
+            try {
+                NativeLibraryLoader.load(libName, SSL.class.getClassLoader());
+                loaded = true;
+                break;
+            } catch (Throwable t) {
+                lastException = t;
+                continue;
+            }
+        }
+
+        if (!loaded) {
+            Throwables.propagate(lastException);
+        }
+
+        try {
+            overrideExceptionValue();
+            Library.initialize("provided");
+            SSL.initialize(null);
+        } catch (Throwable t) {
+            Throwables.propagate(t);
+        }
+    }
+
+    // Netty hardcodes the dependency to the Openssl class in such a way that it makes it difficult
+    // to load tcnative outside netty.
+    // This overrides netty's Openssl availablity to always report success since we are taking care of loading
+    // the library.
+    private static void overrideExceptionValue() {
+        Field exceptionField = null;
+        Field modifiersField = null;
+        int originalModifiers = 0;
+        boolean setModifiers = false;
+
+        try {
+            exceptionField = OpenSsl.class.getDeclaredField("UNAVAILABILITY_CAUSE");
+            exceptionField.setAccessible(true);
+
+            modifiersField = exceptionField.getClass().getDeclaredField("modifiers");
+            modifiersField.setAccessible(true);
+            originalModifiers = modifiersField.getInt(exceptionField);
+            modifiersField.setInt(exceptionField, originalModifiers & ~Modifier.FINAL);
+            setModifiers = true;
+            // Override the default value. This will trigger the static block to run which tries to load the libs,
+            // however we know that will fail.
+            exceptionField.set(null, null);
+        } catch (Throwable t) {
+            Throwables.propagate(t);
+        } finally {
+            // Restore the fields to their original values
```

Comment:
This is not needed. The accessible flag is for the local accessor which is gone when this method returns